openlibrary/tests/accounts/test_models.py

```diff
@@ -1,7 +1,28 @@
-from openlibrary.accounts import model
+from openlibrary.accounts import model, InternetArchiveAccount
+from requests.models import Response
+
 
 def test_verify_hash():
       secret_key = "aqXwLJVOcV"
       hash = model.generate_hash(secret_key, "foo")
       assert model.verify_hash(secret_key, "foo", hash) == True
 
+
+class TestInternetArchiveAccount:
+      def test_xauth_error_handling_no_json(self, monkeypatch):
+            xauth = InternetArchiveAccount.xauth
+
+            resp = Response()
+            resp.status_code = 500
+            resp._content = b'Internal Server Error'
+            monkeypatch.setattr(model.requests, 'post', lambda url, **kwargs: resp)
+            assert xauth('create', s3_key='_', s3_secret='_') == {'code': 500, 'error': 'Internal Server Error'}
+
+      def test_xauth_error_handling_has_json(self, monkeypatch):
```

Comment:
:+1: for tests, but after reviewing the code, this particular test is actually testing whether `xauth()` passes through JSON untouched  regardless of the response code.   I have no idea whether this is reasonable 400  response from the external API.   Any response code and any JSON body is equivalent here.  The test and test implies the code it doing more that it is.

Reply:
:+1: Renamed